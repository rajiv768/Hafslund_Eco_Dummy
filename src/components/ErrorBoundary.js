import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ErrorMessage from './ErrorMessage';

class ErrorBoundary extends Component {
  state = {
    error: null,
    errorInfo: null,
  };

  componentDidCatch(error, errorInfo) {
    console.error(error);
    console.error(errorInfo);
    this.setState({ error, errorInfo });
  }

  render() {
    if (this.state.error) {
      return <ErrorMessage />;
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.element.isRequired,
};

export default ErrorBoundary;
