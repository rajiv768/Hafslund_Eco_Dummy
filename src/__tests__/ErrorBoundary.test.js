import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { StateMock } from '@react-mock/state';
import '@testing-library/jest-dom/extend-expect';

import ErrorBoundary from '../components/ErrorBoundary';

afterEach(cleanup);

// https://react-testing-examples.com/jest-rtl/local-state/
describe('<ErrorBoundary />', () => {
  it('renders the app without error', () => {
    const { container, getByText } = render(
      <StateMock state={{ error: false }}>
        <ErrorBoundary>
          <div>App</div>
        </ErrorBoundary>
      </StateMock>,
    );

    expect(getByText('App')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders err message when there is error', () => {
    const { container, getByText } = render(
      <StateMock state={{ error: true }}>
        <ErrorBoundary>
          <div>App</div>
        </ErrorBoundary>
      </StateMock>,
    );

    expect(getByText('Sorry â€” something has gone wrong.')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
