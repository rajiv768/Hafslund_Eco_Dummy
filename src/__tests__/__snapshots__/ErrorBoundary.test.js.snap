// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders err message when there is error 1`] = `
<div>
  <p>
    Sorry â€” something has gone wrong.
  </p>
</div>
`;

exports[`<ErrorBoundary /> renders the app without error 1`] = `
<div>
  App
</div>
`;
